frontend/src/components/UI/AllBLIRow/AllBLIRow.js

```diff
@@ -0,0 +1,207 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useSelector } from "react-redux";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faClock } from "@fortawesome/free-regular-svg-icons";
+import CurrencyFormat from "react-currency-format";
+import TableTag from "../TableTag";
+import ChangeIcons from "../ChangeIcons";
+import TableRowExpandable from "../TableRowExpandable";
+import {
+    loggedInName,
+    formatDateNeeded,
+    formatDateToMonthDayYear,
+    displayFeePercent,
+    totalBudgetLineFeeAmount,
+    totalBudgetLineAmountPlusFees,
+} from "../../../helpers/utils";
+
+/**
+ * BLIRow component that represents a single row in the Budget Lines table.
+ * @param {Object} props - The props for the BLIRow component.
+ * @param {Object} props.bl - The budget line object.
+ * @param {boolean} [props.canUserEditBudgetLines] - Whether the user can edit budget lines.
+ * @param {boolean} [props.isReviewMode] - Whether the user is in review mode.
+ * @param {Function} [props.handleSetBudgetLineForEditing] - The function to set the budget line for editing.
+ * @param {Function} [props.handleDeleteBudgetLine] - The function to delete the budget line.
+ * @param {Function} [props.handleDuplicateBudgetLine] - The function to duplicate the budget line.
+ * @param {boolean} [props.readOnly] - Whether the user is in read only mode.
+ * @returns {React.JSX.Element} The BLIRow component.
+ **/
+const AllBLIRow = ({
+    bl: budgetLine,
+    canUserEditBudgetLines,
+    handleSetBudgetLineForEditing = () => {},
+    handleDeleteBudgetLine = () => {},
+    readOnly = false,
+}) => {
+    const [isExpanded, setIsExpanded] = React.useState(false);
+    const [isRowActive, setIsRowActive] = React.useState(false);
+    let loggedInUser = useSelector((state) => loggedInName(state?.auth?.activeUser));
+    const loggedInUserId = useSelector((state) => state?.auth?.activeUser?.id);
+
+    const isBudgetLineDraft = budgetLine?.status === "DRAFT";
```

Comment:
would be nice to create a reusable function for this (I think it's implemented on a few components now)